internal/handlers/pg/msg_find.go

```diff
@@ -110,9 +110,24 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 	return &reply, nil
 }
 
+// fetchParams is used to pass parameters to fetchAndFilterDocs.
+type fetchParams struct {
+	tx              pgx.Tx
+	qp              *pgdb.QueryParams
+	disablePushdown bool
+}
+
 // fetchAndFilterDocs fetches documents from the database and filters them using the provided sqlParam.Filter.
-func (h *Handler) fetchAndFilterDocs(ctx context.Context, tx pgx.Tx, sqlParam *pgdb.QueryParam) ([]*types.Document, error) {
-	iter, err := pgdb.QueryDocuments(ctx, tx, sqlParam)
+func fetchAndFilterDocs(ctx context.Context, fp *fetchParams) ([]*types.Document, error) {
+	// filter is used to filter documents on the FerretDB side,
+	// qp.Filter is used to filter documents on the PostgreSQL side (query pushdown).
+	filter := fp.qp.Filter
+
+	if fp.disablePushdown {
+		fp.qp.Filter = nil
+	}
```

Comment:
Do you need a unit test for this and tigris? Integration test isn't in scope as I read the issue.

Reply:
I believe not for now. When we set `disable-pushdown` flag with running tests, there will be checks (implemented in a separate PR).